runtime/lua/vim/lsp/extensions.lua

```diff
@@ -0,0 +1,50 @@
+local vim = vim
+local validate = vim.validate
+local log = require 'vim.lsp.log'
+local util = require 'vim.lsp.util'
+local buf = require 'vim.lsp.buf'
+
+local M = {}
+
+-- copied from buf.lua
+local function request(method, params, callback)
```

Comment:
I was suggesting you just make this `request_peek` and remove the callback parameter, and then use that throughout this file. 

Remove the copied from buf.lua as well then when you do that.